.cirrus.yml

```diff
@@ -0,0 +1,49 @@
+task:
+  trigger_type: manual
+  name: E2E/Integration Tests
+  timeout_in: 45
+  container:
+    image: ubuntu:21.04
+    kvm: true # required for E2E/Integrations tests
+    cpu: 4
+    memory: 8G
+  env:
+    DEBIAN_FRONTEND: noninteractive
+    CI: true
+    FORCE_COLOR: 1 # force terminal color
+    CIRRUS_CLONE_DEPTH: 0 # Required to get app version
+    PLAYWRIGHT_SKIP_BROWSER_DOWNLOAD: 1 # disable auto-browser download
+    KEYRING: /usr/share/keyrings/nodesource.gpg
+    NODE_VERSION: node_14.x # set node version e.g: "node_16.x" for Node 16 LTS
+    DISTRO: hirsute # release codename for Ubuntu 21.04 - see "lsb_release -s -c"
+  install_deps_script: |
+    apt-get update
+    apt-get install -y --no-install-recommends ca-certificates gcc g++ curl git golang openssh-client make netcat sudo vim gpg \
+                    libgtk2.0-0 libgtk-3-0 libgbm-dev libnotify-dev libgconf-2-4 libnss3 libxss1 libasound2 libxtst6 xauth xvfb
+    curl -sLf -o /dev/null "https://deb.nodesource.com/"$NODE_VERSION"/dists/"$DISTRO"/Release"
+    curl -s https://deb.nodesource.com/gpgkey/nodesource.gpg.key | gpg --dearmor | tee "$KEYRING" >/dev/null
+    echo "deb [signed-by="$KEYRING"] https://deb.nodesource.com/"$NODE_VERSION" "$DISTRO" main" > /etc/apt/sources.list.d/nodesource.list
+    echo "deb-src [signed-by="$KEYRING"] https://deb.nodesource.com/"$NODE_VERSION" "$DISTRO" main" >> /etc/apt/sources.list.d/nodesource.list
+    apt-get update && apt-get install -y  nodejs && node -v && npm -v
+
+  # Tests should run as non-root user - setting up a test user
+  prepate_user_script:
+    - groupadd -g $(stat -c '%g' /dev/kvm) kvm
+    - useradd -m -G kvm ranchertest
+    - cp -rf . /home/ranchertest
+    - chown -R ranchertest.ranchertest /home/ranchertest
+
+  node_modules_cache:
+    folder: /home/ranchertest/node_modules
+    populate_script:
+      - sudo -iu ranchertest DEBUG=pw:install npm ci
```

Comment:
Is the `DEBUG` for playwright? Please add a comment.

Reply:
Hmm, `npm ci` is _only_ in the cache block, so it will never be run on cache hit, right? So it won't, for example, download the `…/resources/*` files (done in the postinstall script).

It's reasonable to argue that we shouldn't be doing it in a postinstall script; however, doesn't that still mean that the build should be broken as-is? Why is it passing?

Reply:
> Is the `DEBUG` for playwright? Please add a comment.

This is an option to watch playwright installing verbosely - it helps to make sure everything is download and it will save time if something goes wrong... instead wait the entire test suite fails and then realize one of the package didn't install.

> Hmm, `npm ci` is _only_ in the cache block, so it will never be run on cache hit, right? So it won't, for example, download the `…/resources/*` files (done in the postinstall script).
It will download+install all deps/plugins and it will store the cache after the first full execution (pass or fail). If something changes (with or without `fingerprint_script` or `key`) , Cirrus CI will upload a new cache running `npm ci`.
 
```bash
After the last script instruction for the task succeeds, Cirrus CI will calculate checksum of the cached folder (note that it's unrelated to fingerprint_script or fingerprint_key fields) and re-upload the cache if it finds any changes. To avoid a time-costly re-upload, remove volatile files from the cache (for example, in the last script instruction of a task).
```


> 
> It's reasonable to argue that we shouldn't be doing it in a postinstall script; however, doesn't that still mean that the build should be broken as-is? Why is it passing?
No clue.

Reply:
Please add a comment _in the file_, not on GitHub :)

Also, short arguments here too. `sudo -iu` in this file should generally be `sudo --login --user=…`, etc.